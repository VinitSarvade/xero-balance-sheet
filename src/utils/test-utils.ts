import {
  format as prettyFormat,
  plugins as prettyFormatPlugins,
} from "pretty-format";
import { http, HttpResponse } from "msw";
import { BalanceSheetReport } from "@/modules/xero/xeroReport.dto";

export function format(value: any) {
  return prettyFormat(value, {
    escapeRegex: true,
    indent: 2,
    plugins: [
      prettyFormatPlugins.ReactTestComponent,
      prettyFormatPlugins.ReactElement,
      prettyFormatPlugins.DOMElement,
      prettyFormatPlugins.DOMCollection,
      prettyFormatPlugins.Immutable,
      prettyFormatPlugins.AsymmetricMatcher,
    ],
    printFunctionName: false,
  });
}

export const BalanceSheetMock = {
  ReportID: "report-id",
  ReportName: "Balance Sheet",
  ReportDate: "2022-01-01",
  ReportTitles: ["Balance Sheet", "Test Company"],
  Rows: [
    {
      RowType: "Header",
      Cells: [{ Value: "" }, { Value: "2023" }, { Value: "2024" }],
    },
    {
      RowType: "Section",
      Title: "Assets",
      Rows: [],
    },
    {
      RowType: "Section",
      Title: "Bank",
      Rows: [
        {
          RowType: "Row",
          Cells: [
            { Value: "My Bank Account" },
            { Value: "126.70" },
            { Value: "99.60" },
          ],
        },
        {
          RowType: "Row",
          Cells: [
            { Value: "Sample Business" },
            { Value: "92911.00" },
            { Value: "92911.00" },
          ],
        },
        {
          RowType: "Row",
          Cells: [
            { Value: "Sample Business 1" },
            { Value: "11039.00" },
            { Value: "11039.00" },
          ],
        },
        {
          RowType: "SummaryRow",
          Cells: [
            { Value: "Total Bank" },
            { Value: "104076.70" },
            { Value: "104049.60" },
          ],
        },
      ],
    },
    {
      RowType: "Section",
      Title: "Liabilities",
      Rows: [],
    },
    {
      RowType: "Section",
      Title: "Current Liabilities",
      Rows: [
        {
          RowType: "Row",
          Cells: [
            { Value: "Car Loan" },
            { Value: "8000.00" },
            { Value: "8000.00" },
          ],
        },
      ],
    },
    {
      RowType: "Section",
      Title: "",
      Rows: [
        {
          RowType: "SummaryRow",
          Cells: [
            { Value: "Total Current Liabilities" },
            { Value: "8000.00" },
            { Value: "8000.00" },
          ],
        },
      ],
    },
  ],
} as BalanceSheetReport;

export function getHandlers() {
  return [
    http.get(`${process.env.BASE_URL}/api/xero/balance-sheet`, () => {
      return HttpResponse.json(BalanceSheetMock);
    }),
  ];
}
