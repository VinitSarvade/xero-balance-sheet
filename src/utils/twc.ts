import { createTwc } from "react-twc";
import { cn } from "./cn";

export const twc = createTwc({ compose: cn });
