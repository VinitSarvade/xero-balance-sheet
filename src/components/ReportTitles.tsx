import { twc } from "@/utils/twc";

interface ReportTitlesProps {
  titles: string[];
}

const H1 = twc.h1`text-4xl`;
const H2 = twc.h2`mt-2 text-2xl tracking-tight text-zinc-700`;

export default function ReportTitles({ titles }: ReportTitlesProps) {
  return titles.map((title, idx) => {
    const Title = idx === 0 ? H1 : H2;
    return (
      <Title key={title} className="text-center font-semibold">
        {title}
      </Title>
    );
  });
}
