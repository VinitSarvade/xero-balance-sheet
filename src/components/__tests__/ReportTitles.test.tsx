/// <reference lib="dom" />
import { expect, describe, it, afterEach } from "bun:test";
import { render, cleanup } from "@testing-library/react";
import { format } from "@/utils/test-utils";
import ReportTitles from "../ReportTitles";

describe("ReportTitles", () => {
  it("renders titles with correct heading levels", () => {
    const titles = ["Title 1", "Title 2", "Title 3"];
    const { getAllByRole } = render(<ReportTitles titles={titles} />);

    const headings = getAllByRole("heading");
    expect(headings).toHaveLength(3);
    expect(headings[0].textContent).toBe(titles[0]);
    expect(headings[0].tagName).toBe("H1");
    expect(headings[1].textContent).toBe(titles[1]);
    expect(headings[1].tagName).toBe("H2");
    expect(headings[2].textContent).toBe(titles[2]);
    expect(headings[2].tagName).toBe("H2");
  });

  it("renders no titles when empty array is passed", () => {
    const { queryAllByRole } = render(<ReportTitles titles={[]} />);
    const headings = queryAllByRole("heading");
    expect(headings).toHaveLength(0);
  });

  it("matches snapshot", () => {
    const titles = ["Title 1", "Title 2", "Title 3"];
    const { asFragment } = render(<ReportTitles titles={titles} />);
    expect(format(asFragment())).toMatchSnapshot();
  });
});
