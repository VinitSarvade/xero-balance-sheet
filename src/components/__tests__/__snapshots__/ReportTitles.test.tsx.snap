// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTitles matches snapshot 1`] = `
"<DocumentFragment>
  <h1
    class="text-4xl text-center font-semibold"
  >
    Title 1
  </h1>
  <h2
    class="mt-2 text-2xl tracking-tight text-zinc-700 text-center font-semibold"
  >
    Title 2
  </h2>
  <h2
    class="mt-2 text-2xl tracking-tight text-zinc-700 text-center font-semibold"
  >
    Title 3
  </h2>
</DocumentFragment>"
`;
